src/recipe/test/unit/recipe.service.spec.ts

```diff
@@ -17,170 +21,104 @@ const newRecipe = {
 
 describe('Recipe Service', () => {
   let recipeService: RecipeService;
+  let commandBus: CommandBus;
+  let queryBus: QueryBus;
+  let userService: UserService;
+  let testUser: User;
   let prismaService: PrismaService;
 
-  beforeEach(async () => {
+  beforeAll(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [RecipeService, PrismaService],
+      imports: [AppModule, CqrsModule],
     }).compile();
 
+    await module.createNestApplication().init();
     recipeService = module.get<RecipeService>(RecipeService);
+    commandBus = module.get<CommandBus>(CommandBus);
+    queryBus = module.get<QueryBus>(QueryBus);
+    userService = module.get<UserService>(UserService);
     prismaService = module.get<PrismaService>(PrismaService);
+
+    testUser = await userService.generateAccount(
+      process.env.TEST_NAME,
+      process.env.TEST_PASSWORD,
+      Role.USER,
+    );
   });
 
-  it('should create a recipe', async () => {
-    const mockResult = {
-      id: 1,
-      authorId: 1,
-      name: 'Dumplings',
-      description: 'Easy dumplings recipe',
-      imageURL: 'imageURL',
-      preparing: [{ id: 1, step: 'add flour', order: 1 }],
-      ingredients: [{ id: 1, name: 'flour', amount: 'spoon' }],
-    } as any;
+  afterEach(async () => {
+    jest.clearAllMocks();
+    await prismaService.recipe.deleteMany();
```

Comment:
keep in mind that performing that kind of operation after each test case is time-consuming. 
On your local machine, it would not be a problem but running many tests can cause issues.

What would be a good idea is to prepare some test data set once per test file. This data set should include all things that your test cases would use.

Once the all test cases are done then you could just remove the database.
Example setup could include:
- using different database for tests and development
- running migration that creates `test` database when tests start
- removing this `test` database when test cases are done